External/FEXCore/Source/Interface/Core/JIT/Arm64/JIT.cpp

```diff
@@ -1029,12 +1045,18 @@ void JITCore::CreateCustomDispatch(FEXCore::Core::InternalThreadState *Thread) {
     b(&NoBlock, Condition::ne);
 
     // Now load the actual host block to execute if we can
-    ldr(x0, MemOperand(x0, offsetof(FEXCore::BlockCache::BlockCacheEntry, HostCode)));
-    cbz(x0, &NoBlock);
+    ldr(x3, MemOperand(x0, offsetof(FEXCore::BlockCache::BlockCacheEntry, HostCode)));
+    cbz(x3, &NoBlock);
 
     // If we've made it here then we have a real compiled block
     {
-      blr(x0);
+      // update L1 cache
+      LoadConstant(x0, Thread->BlockCache->GetL1Pointer());
+
+      and_(x1, RipReg, 1 * 1024 * 1024 - 1);
```

Comment:
Don't use hard coded L1C size.
Pull a constant from the BlockCache class instead

Reply:
Done